src/rust/engine/src/rule_graph.rs

```diff
@@ -837,8 +820,10 @@ fn rhs_for_select(tasks: &Tasks, subject_type: TypeId, select: &Select) -> Entri
   if externs::satisfied_by_type(&select.product, &subject_type) {
     // NB a matching subject is always picked first
     vec![Entry::new_subject_is_product(subject_type)]
+  } else if let Some(&(ref key, _)) = tasks.gen_singleton(&select.product) {
```

Comment:
@baroquebobcat : The reason it works now is that we install dummy copies of those tasks from the python side (the `*_noop` tasks in `fs.py`).